Restructure scraping settings

gitgiggety opened the pull request:
As discussed in #985 and [commented](https://github.com/stashapp/stash/issues/985#issuecomment-871083394) by @WithoutPants I moved the scraping configuration from the (general) Configuration tab to the Scrapers tab and renamed it to Scraping. The scrapers are now wrapped in collapsible so hidden by default.

The end result looks like this:
![image](https://user-images.githubusercontent.com/79809426/124313909-d833e880-db71-11eb-9857-e4b2c4f36280.png)

jeremymeyers commented:
This looks great though personally i would move the "Save" above the scrapers section since there is no interaction to save in the scrapers section.

jeremymeyers commented:
Also might be an opportunity to implement #1446  (if people think its a good idea)

gitgiggety commented:
During implementation I did think for a second whether the Save button should be in between or at the bottom. But for one all (settings) pages have it at the bottom, so it would be quite strange to put it somewhere else on this tab. Secondly the button isn't really "hidden" IMO as the page obviously is a lot shorter now with the scraper lists being collapsed by default. So on any (desktop) screen the button is in view when you open this tab. And on mobile you would be scrolling anyway, and I guess the save button would be quite quickly into view then too, as the collapsed scraper lists would easily fit onto the screen.

And lastly, the feature request you're linking to would warrant the need for saving the scrappers, so at that point the Save button would still have to be at the bottom anyway.

WithoutPants commented on graphql/documents/data/config.graphql:
```diff
@@ -32,9 +32,6 @@ fragment ConfigGeneralData on ConfigGeneralResult {
   galleryExtensions
   excludes
   imageExcludes
-  scraperUserAgent
-  scraperCertCheck
-  scraperCDPPath
```
For compatibility purposes, we want to avoid removing these right now. They should be marked as deprecated, but should still function correctly.

gitgiggety replied:
I'm not too familiar with GraphQL and the usage in Stash. But aren't these fragments only used by the client side code? (in other words: used by the code generator for React). So removing these would then be fine I guess as long as the web interface doesn't use them? (which obviously isn't the case anymore)

gitgiggety replied:
@WithoutPants any comment on this?

I did accidentally revert it as well, but as stated in aboves comment I'm not entirely sure if it was needed. This because I believe it's only used by the client side code (generator). So if that's the case different clients of the API aren't using this so removing it would be fine as the Stash web UI doesn't need it anymore.

WithoutPants replied:
Oh, yes, you are correct. If it's client side it can be removed assuming it's unused.

gitgiggety replied:
> If it's client side

Well, I'm not sure (either) if it is client side only :) But for now I left it in. But feel free to remove it if you think it is unnecessary.

WithoutPants commented on graphql/schema/types/config.graphql:
```diff
@@ -89,12 +89,6 @@ input ConfigGeneralInput {
   excludes: [String!]
   """Array of file regexp to exclude from Image Scans"""
   imageExcludes: [String!]
-  """Scraper user agent string"""
-  scraperUserAgent: String
-  """Scraper CDP path. Path to chrome executable or remote address"""
-  scraperCDPPath: String
-  """Whether the scraper should check for invalid certificates"""
-  scraperCertCheck: Boolean!
```
As above, these should be left in. `scraperCertCheck` should be made optional. For these three fields, the server should only change the configuration value if the fields are present. Mark as deprecated as needed.

WithoutPants commented on graphql/schema/types/config.graphql:
```diff
@@ -162,12 +156,6 @@ type ConfigGeneralResult {
   excludes: [String!]!
   """Array of file regexp to exclude from Image Scans"""
   imageExcludes: [String!]!
-  """Scraper user agent string"""
-  scraperUserAgent: String
-  """Scraper CDP path. Path to chrome executable or remote address"""
-  scraperCDPPath: String
-  """Whether the scraper should check for invalid certificates"""
-  scraperCertCheck: Boolean!
```
As above - leave in and deprecate.

WithoutPants commented on ui/v2.5/src/locales/en-GB.json:
```diff
@@ -240,9 +240,10 @@
       "triggers_on": "Triggers on"
     },
-    "scrapers": {
+    "scraping": {
```
This change should be added to the other locale files as well.

WithoutPants commented on ui/v2.5/src/locales/en-GB.json:
```diff
@@ -151,7 +151,7 @@
       "logs": "Logs",
       "plugins": "Plugins",
-      "scrapers": "Scrapers",
+      "scraping": "Scraping",
```
Please remove `scrapers` from the other locale files.

gitgiggety replied:
As internationalization is new I'm wondering how you would want to handle this. Do you want me to remove these lines from the other locale files, or change them into `"scraping": ""`? So updating the key, and making the value an empty string.
Adding (or updating) translations using an empty string IMO has the advantage that translators can find missing translations more easily and all files could be kept identical in format (so every file would have `"scraping"` on line 154 for example) which I think would make it easier to "compare" the files.

WithoutPants replied:
This way would need some modification to our message loading code. Currently, we merge the chosen locale messages with the default message locale (`en-GB`), using lodash's `merge` function. This combines the chosen and default maps, using the default value if the chosen value resolves to `undefined`. You will probably need to use `mergeWith` supplying a customiser function to return the default value if the chosen value is empty.

I'm also not completely sold on this approach as it makes the `en-US` locale unnecessarily verbose (most of the messages should use the `en-GB` value). I'd prefer not to have to maintain identical locale file structures as well.

However, when adding new messages to add a new placeholder message to the other locales, I think it's not a bad approach to add an empty string placeholder value - provided the above code change is made.

gitgiggety replied:
So if I do understand you correctly you _do_ want me to set the value / translation to `""` and update the translation loading code to ignore these values?

WithoutPants replied:
Sorry, was offering an opinion rather than a directive. :)

If you're happy to do the changes to the message loading code, then go ahead and do it. Otherwise, just remove `scrapers` from the other json locale files.

gitgiggety replied:
That clears it up, I will make the changes then.

I was a bit unsure because the first part of the comment seemed to imply you were totally against using `""` and at the end you seemed to find it a good solution (if the fix to merging is added as well, obviously).

WithoutPants replied:
Yeah I can understand why it was confusing. What I wasn't keen on was trying to maintain identical file structures for all of the locale files.

gitgiggety commented:
Rebased on current develop and updated excepted for the two things I commented on / asked a question about.

gitgiggety commented:
Rebased on top of current develop branch to include pt-BR as well.

Also added support for language lines to be empty and fall back to the `en-GB` line. But in the end it turned out that all languages already had a translation for `Scraping` which I just reused / copied over.
